### DISCUSSION TITLE
Issues with Expand param

### DISCUSSION THREAD

**Discussion by dustinknopoff** [pocketbase/pocketbase#54] [Q&A] [answered]
I think I might have found an issue with using the `expand` query param

I've created a super simple pocketbase. 

- Tags, Authors, and a Post collection which has one or more tags and authors

```sql
CREATE TABLE `tag` (`created` TEXT DEFAULT "" NOT NULL, `id` TEXT PRIMARY KEY, `name` TEXT DEFAULT '', `updated` TEXT DEFAULT "" NOT NULL);
CREATE INDEX `tag_created_idx` ON `tag` (`created`);
CREATE TABLE `author` (`created` TEXT DEFAULT "" NOT NULL, `id` TEXT PRIMARY KEY, `name` TEXT DEFAULT '', `updated` TEXT DEFAULT "" NOT NULL);
CREATE INDEX `author_created_idx` ON `author` (`created`);
CREATE TABLE `post` ("authors" TEXT DEFAULT '', `created` TEXT DEFAULT "" NOT NULL, `id` TEXT PRIMARY KEY, "tags" TEXT DEFAULT '', "title" TEXT DEFAULT '', `updated` TEXT DEFAULT "" NOT NULL);
CREATE INDEX `post_created_idx` ON `post` (`created`);
```
<img width="630" alt="Screen Shot 2022-07-10 at 12 35 58 PM" src="https://user-images.githubusercontent.com/23691663/178143185-764452ed-8dde-43d4-a32d-972858ea4c1b.png">

When I go to fetch it with the expanded fields, There's nothing in the `@expand` key

```js
const PocketBase = require('pocketbase')

const client = new PocketBase('http://localhost:8090')

client.Records.getFullList('post', { expand: 'tags,authors' }).then(console.log)
```

Outputs

```js
[
  e {
    id: 'MTbwIWI2RxDAzH1',
    created: '2022-07-10 11:23:51.870',
    updated: '2022-07-10 11:23:51.870',
    '@collectionId': '5HIxphRwA39S1pG',
    '@collectionName': 'post',
    authors: [ 'ShBveSUY5Z1gMrs' ],
    tags: [ 'TFr6RQVU7jkfJN6' ],
    title: 'Example 1',
    '@expand': {}
  }
]
```

**Comment by ganigeorgiev** [comment, marked as answer]
Could you check the viewRule of your related collections?

This is not documented (I'll updated it), but PocketBase will not expand records if the user doesn't have access to view them (https://github.com/pocketbase/pocketbase/blob/master/apis/record.go#L406-L432). If you are running the application in debug mode (aka. `--debug`), you may see a log error printed in the console.

**Reply by dustinknopoff** [reply]
I've set it now for all the collections to allow listing and viewing for everyone and it remains.


(The --debug output now)

```
[0.00ms] SELECT {{_collections}}.* FROM `_collections` WHERE (`id`='post') OR (`name`='post') LIMIT 1
[0.00ms] SELECT count(*) FROM `post`
[0.00ms] SELECT `post`.* FROM `post` LIMIT 30
```

**Reply by ganigeorgiev** [reply]
I've justed it locally, with both single and multiple value fields and it works as expected.
But I noticed that in your call to getFullList you are passing the query parameters as 2nd argument, while it should be the 3rd, eka.:
```js
client.Records.getFullList('post', 100, { expand: 'tags,authors' }).then(console.log)
```
_ the second argument, 100, is the batch size, aka. how many items in a single request to fetch_

**Reply by dustinknopoff** [reply]
Awesome! That fixed it. Thankyou!
